### PR TITLE
Adding apiman plugin

### PR DESCRIPTION
**Pull request by KurtStam**
@jstrachan, @jdyson can you review that this is how setting the config works?

### CODE REVIEW

**Comment by jimmidyson** [comment]
You'll need to add the config file flag to the dockerfile too. -c iirc

File: apiman/config.js.tmpl [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    },
+    "ui" : {
+        "header" : false
+    },
+    "api" : {
+        "endpoint" : "{{ .Env.APIMAN_MANAGER_API_ENDPOINT : APIMANMANAGERAPI:8998 }}",
+        "auth" : {
+            "type" : "basic",
+            "basic" : {
+                "username" : "admin",
```
Comment (jimmidyson) [inline apiman/config.js.tmpl]: Hard coded username/password? Env vars please.

File: gulpfile.js [hunk 2]
```diff
@@ -318,6 +318,17 @@ gulp.task('connect', ['watch'], function() {
     res.set('Content-Type', 'application/javascript');
     res.send(configJs);
   });
+  //TODO check up with jdyson to figure out how this works on prod
+  hawtio.use('/apiman/config.js', function(req, res, next) {
+    var configJs = 'window.APIMAN_CONFIG = {' +
+      ' auth: {' +
```
Comment (jimmidyson) [inline gulpfile.js]: I assume you meant to use the APIMAN_CONFIG content in here rather than OS auth config?

**Comment by jimmidyson** [comment]
@gashcrumb - seeing a need for a hawtio-config service that plugins can register & retrieve config through. What do you think?

**Comment by jstrachan** [comment]
@jimmidyson you mean for server side config rather than user preferences stuff right? I guess any server side service could expose a REST API to expose any old config stuff it needed? I wonder if there should be a way to expose to consoles a subset of env vars or something?

**Comment by jimmidyson** [comment]
@jstrachan - what i mean is that plugins could inject config files to be retrieved via an angular service. That would make it easier for plugins to extend config for example. Then config can be injected directly into controllers, etc as required.

Right now we're polluting global scope (e.g. window.OS_CONFIG) which isn't cool & could lead to problems in future.

**Comment by jstrachan** [comment]
@jimmidyson isn't that just the same as plugins creating an angular service that any other plugin who's interested could get injected? Then custom plugins which extend other plugins have the choice of reusing that config or overwriting it with a different implementation? 

e.g. apimanConfig could be a service that any angularjs service can get injected; no need for a global variable etc

**Comment by jimmidyson** [comment]
What I'm trying to say (badly) is that a common service to handle plugin configs would be nice rather than each plugin implementing it in their own way. Just a way for a plugin to say this is the config file to load, & interested plugins to say give me this config please.

**Comment by gashcrumb** [comment]
Agreed, it'd be best if we had a fairly common pattern of how to fetch config and an easy way to embed a simple json/js file that can be set via environment variables that hawtio-node-backend can serve out without having to add in a custom middleware any time a new plugin has some needed config.

On the frontend side for fetching that config I've used hawtioPluginLoader.registerPreBootstrapTask() as you can fetch whatever you need async before angular bootstraps, which leads to less pain later on, otherwise you're likely going to deal with a service that's going to be returning a lot of promises and have a lot of typical async issues.  Now, for openshift I did pick that config.js specifically because that's how they do it in the openshift console, it allows us to easily use their oauth config.  But yeah, probably be better if plugins either just used a .json file and not a .js so they don't have to use $.getScript(), or if a .js file is used, set the config in the same module name as the plugin code to avoid clashes.

Though to be honest, the global namespace is pretty filled up as it is, I wouldn't worry too much about polluting it :-)

**Comment by gashcrumb** [comment]
BTW, I did mention to Kurt to get what he could in, I'll look into extending hawtio-node-backend so it's easy to register .js/.json files to be injected with environment variables later.

**Comment by jimmidyson** [comment]
Agreed I'd merge this once Kurt is happy it's working & we can work on a config service later?

**Comment by gashcrumb** [comment]
Yup!

File: gulpfile.js (hunk 2 above)
Comment (KurtStam) [inline gulpfile.js]: You tell me :). I want it to participate in the SSO session of the hawtio console.

Comment (gashcrumb) [inline gulpfile.js]: If you use $http for your requests then hawtio-oauth automatically adds the
token as a header, so you should be all set there, at least client-side...

**Comment by KurtStam** [comment]
I think this is good to go now.

**Comment by gashcrumb** [comment]
In `apiman.css` (which I think gets inserted via bower) there's this rule:

```
.navbar-default {
  background-color: #0B61A4;
  border-bottom: 1px solid #FF9200;
}
```

which causes the fabric8-console's nav header to have a blue background and the sub-tab header to have a yellow bottom border.  Maybe in apiman you could split the css rules into two files, one exported in the plugin for the views, and another used locally when it's a standalone thing?

**Comment by gashcrumb** [comment]
Also there doesn't seem to be any isValid() function for apiman's nav bar entry, so it's always visible...  I thought you added one, am I just running with an old version?

**Comment by KurtStam** [comment]
Thanks Stan, I will talk to the apiman guys who are adding the isValid entry in the new apiman bower component version so hopefully we can roll in the style fix.

**Comment by EricWittmann** [comment]
I will fix up the CSS issue - that's no problem.  The solution to the isValid problem is here:

https://github.com/apiman/apiman/blob/master/manager/ui/hawtio/plugins/api-manager/ts/apimanPlugin.ts#L144-L155

Is that OK?  Or do we specifically need isValid()?

**Comment by gashcrumb** [comment]
@KurtStam thanks!

@EricWittmann yeah, think that'll work just fine, thanks!

**Pull request merged**